### PR TITLE
Fixed  #7211, #9197, #7864, [ch15504] - Added additional LDAP fields to sync

### LINKED ISSUE #7211: Sync more attributes via LDAP
**Server (please complete the following information):**
 - Snipe-IT Version: 4.7.5
 - OS: Windows Server 2016
 - Web Server: IIS
 - PHP Version: 7.3.6

**Is your feature request related to a problem? Please describe.**
We have our Active Directory attributes filled out fairly well and would like some more attributes to be synced with Snipe-IT to stop manual input.  Perhaps the ability to choose which attributes to actually pull in would be even better, so if other users have custom attributes, they can sync it without having to change Snipe-IT's code.

**Describe the solution you'd like**
We'd like Title, Manager, Department, Phone, Website, Address, City, State, Country, and Zip to be added.  In my particular environment, we'd actually use `division` for Department, but that can be handled via the mapper.

**Describe alternatives you've considered**
We could work around this by writing directly to the database using the Employee ID as the key.

### PR DESCRIPTION
**Pull request by Godmartinz**
included is 4 fields that now sync with LDAP:
Telephone
Department ID
Country
Job Title

A migration has also been added for the Settings table.

### CODE REVIEW

**Comment by snipe** [comment]
This looks good, thanks! The one thing I'm wondering is if it wouldn't make more sense to include a Department, not department ID - and try to match or generate the Dept name on the fly during that sync. I feel like most people would have the Dept name in the LDAP/AD users directory, not the ID, so if they map that incorrectly, it could cause some confusion. What are your thoughts, @uberbrady?

**Comment by snipe** [comment]
(Also please try to remember to title your PRs to make them easier to generate changelogs from. Something like "Fixed  [ch15504]  - Added additional LDAP fields to sync" makes it a lot easier for me when I have to create changelogs for new versions.   Thanks!)

**Comment by Godmartinz** [comment]
I can do that no problem, I was matching fields from our users' table to AD, I could add another column for department as well. To be honest there are a few other fields that will probably be handy to sync.

**Comment by snipe** [comment]
>  I was matching fields from our users' table to AD, I could add another column for department as well

Well I see department ID in the list of fields, I just mean that in AD, it's more likely that someone will have a Dept name, not an Dept ID, in their directory. So we don't want to store the department name in the user table, we would:

-  map the `department` field in AD
- check the departments table in Snipe-IT to see if it exists already (by name)
- add it and get the new ID if it doesn't, just get the ID if it does (same as we do for the importer)
- add/update the department_id to the user record

Basically, the only difference is that you'd be adding in the step where we check for the department ID in the departments table, instead of expecting the Snipe-IT admins to store the `department_id` as an attribute on the user in their AD/LDAP directory. (And @uberbrady, correct me if I'm wrong here. When I used to manage our AD, I'd never have used a Dept ID, just a Dept name, but my case may have been weird.)

**Comment by uberbrady** [comment]
I agree that finding (or creating!) the Department would be a really nice feature to throw in.

@Godmartinz - check out `firstOrCreate()` - a really neat method that finds the thing with the appropriate values, _or_ creates it for you - will make your code a little bit cleaner.

But it's also completely fine to do it without that helper method, it's just a convenient short-hand.

**Comment by uberbrady** [comment]
@Godmartinz If you wanted to just get this PR off your plate and do Department under a different PR, that'd be fine with me - just let me know.

**Comment by Godmartinz** [comment]
department id has been removed. I'll get the department done and hopefully manager done as well in a seperate PR. @uberbrady @snipe

**Review by uberbrady — APPROVED** [review]
Excellent! Thanks for sticking with it @Godmartinz ! We'll nail Department with your next one.
